### PR TITLE
Docs: Add docs on setting up devd rules to flash MCUs on FreeBSD.

### PR DESCRIPTION
**Pull request by petejohanson**
## Description

* Document how to set up `devd` config blocks that match on vendor
  and product and then update group + permissions. Roughly matches
  the approach taken w/ udev on Linux.

With the changes in place, I can plug in my Kyria, hitting the reset button and as a regular user, inspect the USB device:

```
➜  qmk_firmware git:(docs-faq-build-freebsd-devd-config-flashing) usbconfig
ugen2.3: <ATMEL ATm32U4DFU> at usbus2, cfg=0 md=HOST spd=FULL (12Mbps) pwr=ON (100mA)
```

And even flash it:

```
➜  qmk_firmware git:(docs-faq-build-freebsd-devd-config-flashing) qmk flash -kb kyria/rev1 -km default                                      
Ψ Compiling keymap with gmake kyria/rev1:default:flash


QMK Firmware 0.8.118
Making kyria/rev1 with keymap default and target flash

avr-gcc (GCC) 9.1.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Size before:
   text	   data	    bss	    dec	    hex	filename
      0	  25006	      0	  25006	   61ae	.build/kyria_rev1_default.hex

Copying kyria_rev1_default.hex to qmk_firmware folder                                               [OK]
Checking file size of kyria_rev1_default.hex                                                        [OK]
 * The firmware size is fine - 25006/28672 (87%, 3666 bytes free)
Bootloader Version: 0x00 (0)
Erasing flash...  Success
Checking memory from 0x0 to 0x6FFF...  Empty.
Checking memory from 0x0 to 0x61FF...  Empty.
0%                            100%  Programming 0x6200 bytes...
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>]  Success
0%                            100%  Reading 0x7000 bytes...
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>]  Success
Validating...  Success
0x6200 bytes written into 0x7000 bytes memory (87.50%).
gmake[1]: *** [tmk_core/avr.mk:320: flash] Error 254
Make finished with errors
gmake: *** [Makefile:579: kyria/rev1:default:flash] Error 1
```

(Note: The flash error I still am digging into, it is intermitent, I suspect a cheap USB-A -> USB-C cable, or *possibly* newer avr-gcc, which only vesion 9 is packaged in ports.)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* Previously, FreeBSD users had to flash as root, or manually update device permissions after resetting their keyboards.

## Checklist

- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
I suspect that this may overlap partially with #8750

File: docs/faq_build.md [hunk 1]
```diff
@@ -75,6 +75,87 @@ SUBSYSTEMS=="usb", ATTRS{idVendor}=="0483", ATTRS{idProduct}=="df11", MODE:="066
...
+**/etc/devd/52-tmk-keyboard.conf:**
+```
+# tmk keyboard products     https://github.com/tmk/tmk_keyboard
+notify 1000 {
+        match "system" "USB";
+        match "vendor" "0xfeed";
+        action "chown :dfu /dev/$cdev && chmod 0660 /dev/$cdev";
+};
+```
+
```
Comment (fauxpark) [inline docs/faq_build.md:117]:
```suggestion
```

Comment (petejohanson) [reply docs/faq_build.md:117]: @fauxpark I'm happy to do so, but that certainly will conflict even *more* with the change from #8750 

Any preference on order for doing that?

Comment (fauxpark) [reply docs/faq_build.md:117]: Just thinking out loud. These can probably go in first.

**Pull request closed without merging**